src/extra_checks/checks/model_field_checks.py

```diff
@@ -269,3 +269,40 @@ def apply(
                 hint="Remove `default=None` from field arguments.",
                 obj=field,
             )
+
+
+@registry.register(django.core.checks.Tags.models)
+class CheckFieldChoicesConstraint(CheckModelField):
+    Id = CheckId.X060
+
+    @staticmethod
+    def _repr_choice(value: Any) -> str:
+        if isinstance(value, str):
+            return f'"{value}"'
+        return str(value)
+
+    def apply(
+        self, field: models.fields.Field, field_ast: FieldAST, model: Type[models.Model]
+    ) -> Iterator[django.core.checks.CheckMessage]:
+        if field.choices:
+            field_choices = [c[0] for c in field.choices]
```

Comment:
I think this will fail when using grouped choices. I found that `field.flatchoices` seems to work for me but I don't think this attribute is documented.

I'd also recomment using a `set` since values could potentially be duplicated.

Reply:
Thank you for reminding me about grouped choices, I'll add support for them.

I was using `set` but I want to generate hint with the same order as in choices, will review that too.